strongbox-security/strongbox-user-management/src/main/java/org/carlspring/strongbox/gremlin/adapters/UserAdapter.java

```diff
@@ -0,0 +1,111 @@
+package org.carlspring.strongbox.gremlin.adapters;
+
+import static org.apache.tinkerpop.gremlin.structure.VertexProperty.Cardinality.set;
+import static org.apache.tinkerpop.gremlin.structure.VertexProperty.Cardinality.single;
+import static org.carlspring.strongbox.gremlin.adapters.EntityTraversalUtils.extracPropertytList;
+import static org.carlspring.strongbox.gremlin.adapters.EntityTraversalUtils.extractObject;
+import static org.carlspring.strongbox.gremlin.adapters.EntityTraversalUtils.toLocalDateTime;
+import static org.carlspring.strongbox.gremlin.adapters.EntityTraversalUtils.toLong;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.tinkerpop.gremlin.process.traversal.Traverser;
+import org.apache.tinkerpop.gremlin.structure.Element;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+import org.carlspring.strongbox.db.schema.Vertices;
+import org.carlspring.strongbox.domain.UserEntity;
+import org.carlspring.strongbox.gremlin.dsl.EntityTraversal;
+import org.carlspring.strongbox.gremlin.dsl.__;
+import org.carlspring.strongbox.users.dto.User;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author sbespalov
+ */
+@Component
+public class UserAdapter extends VertexEntityTraversalAdapter<User>
+{
+
+    @Override
+    public Set<String> labels()
+    {
+        return Collections.singleton(Vertices.USER);
+    }
+
+    @Override
+    public EntityTraversal<Vertex, User> fold()
+    {
+        return __.<Vertex, Object>project("id", "uuid", "password", "enabled", "roles", "securityTokenKey", "lastUpdated", "sourceId")
+                 .by(__.id())
+                 .by(__.enrichPropertyValue("uuid"))
+                 .by(__.enrichPropertyValue("password"))
+                 .by(__.enrichPropertyValue("enabled"))
+                 .by(__.enrichPropertyValue("roles"))
+                 .by(__.enrichPropertyValue("securityTokenKey"))
+                 .by(__.enrichPropertyValue("lastUpdated"))
+                 .by(__.enrichPropertyValue("sourceId"))
```

Comment:
Question: Maybe we should extract these to constants? I see they're being used in quite a few places.

Reply:
we have these constants at `org.carlspring.strongbox.db.schema.Properties`, will create a task to replace all the property names literals with constants.

Reply:
[issues/1842](https://github.com/strongbox/strongbox/issues/1842)